components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/CertificateMgtDAO.java

```diff
@@ -180,40 +174,34 @@ public boolean addCertificate(String alias, String endpoint, int tenantId) throw
-            try {
-                connection = APIMgtDBUtil.getConnection();
-                initialAutoCommit = connection.getAutoCommit();
-                connection.setAutoCommit(false);
-                preparedStatement = connection.prepareStatement(addCertQuery);
-                preparedStatement.setInt(1, tenantId);
-                preparedStatement.setString(2, endpoint);
-                preparedStatement.setString(3, alias);
-                result = preparedStatement.executeUpdate() == 1;
-                connection.commit();
+            connection = APIMgtDBUtil.getConnection();
+            initialAutoCommit = connection.getAutoCommit();
+            connection.setAutoCommit(false);
+            preparedStatement = connection.prepareStatement(addCertQuery);
+            preparedStatement.setInt(1, tenantId);
+            preparedStatement.setString(2, endpoint);
+            preparedStatement.setString(3, alias);
+            result = preparedStatement.executeUpdate() == 1;
+            connection.commit();
 
-            } catch (SQLException e) {
-                try {
-                    if (connection != null) {
-                        connection.rollback();
-                    } else {
-                        log.error("Could not perform rollback since the connection is null.");
-                    }
-                } catch (SQLException e1) {
-                    log.error("Error while rolling back the transaction.", e1);
-                }
-                if (log.isDebugEnabled()) {
-                    log.debug("Error occurred while adding certificate metadata to database.", e);
+        } catch (SQLException e) {
+            try {
+                if (connection != null) {
+                    connection.rollback();
+                } else {
+                    log.warn("Could not perform rollback since the connection is null.");
                 }
-                handleException("Error while persisting certificate metadata.", e);
-            } finally {
-                connection.setAutoCommit(initialAutoCommit);
-                APIMgtDBUtil.closeStatement(preparedStatement);
-                APIMgtDBUtil.closeAllConnections(preparedStatement, connection, null);
+            } catch (SQLException e1) {
+                log.error("Error while rolling back the transaction.", e1);
             }
-        } catch (SQLException e) {
             if (log.isDebugEnabled()) {
-                log.debug("Error occurred in finally block while adding certificate metadata. ", e);
+                log.debug("Error occurred while adding certificate metadata to database.", e);
             }
+            handleException("Error while persisting certificate metadata.", e);
+        } finally {
+            APIMgtDBUtil.setAutoCommit(connection, initialAutoCommit);
+            APIMgtDBUtil.closeStatement(preparedStatement);
+            APIMgtDBUtil.closeAllConnections(preparedStatement, connection, null);
```

Comment:
Do we need to explicitly call APIMgtDBUtil.closeStatement? APIMgtDBUtil.closeAllConnections does that as well.